crates/astria-sequencer/src/authority/action.rs

```diff
@@ -94,3 +98,183 @@ impl ActionHandler for SudoAddressChangeAction {
         Ok(())
     }
 }
+
+#[async_trait::async_trait]
+impl ActionHandler for FeeChangeAction {
+    /// check that the signer of the transaction is the current sudo address,
+    /// as only that address can change the fee
+    async fn check_stateful<S: StateReadExt + 'static>(
+        &self,
+        state: &S,
+        from: Address,
+    ) -> Result<()> {
+        // ensure signer is the valid `sudo` key in state
+        let sudo_address = state
+            .get_sudo_address()
+            .await
+            .context("failed to get sudo address from state")?;
+        ensure!(sudo_address == from, "signer is not the sudo key");
+        Ok(())
+    }
+
+    #[instrument(skip_all)]
+    async fn execute<S: StateWriteExt>(&self, state: &mut S, _: Address) -> Result<()> {
+        use crate::{
+            accounts::state_ext::StateWriteExt as _,
+            bridge::state_ext::StateWriteExt as _,
+            ibc::state_ext::StateWriteExt as _,
+            sequence::state_ext::StateWriteExt as _,
+        };
+
+        match self.fee_change {
```

Comment:
do ever get events for this? I wonder if it might be warranted to emit `warn!` level events for each one of these changes.

Reply:
yeah, Jordan and I discussed this, it might be useful to emit ABCI events for certain things occurring for easier indexing in general. this would be one of those spots. will probably look over all of sequencer at some point to decide what to add